src/AppServices/KafkaPage/KafkaPage.test.tsx

```diff
@@ -10,4 +10,30 @@ describe("KafkaPage", () => {
     await waitForI18n(comp);
     expect(comp.baseElement).toMatchSnapshot();
   });
+
+  it("should persist ouiaComponentId of purchase card", async () => {
+    const comp = render(<Example />);
+    await waitForI18n(comp);
+
+    const purchaseCard = comp.getByRole("article", { name: "Purchase now" });
+    expect(purchaseCard.dataset.ouiaComponentId).toBe(
+      "card-overview-purchase-now"
+    );
+
+    const link = comp.getByRole("link", { name: "Purchase now" });
+    expect(link.dataset.ouiaComponentId).toBe("link-purchase-now");
+  });
+
+  it("should persist ouiaComponentId of contact card", async () => {
+    const comp = render(<Example />);
+    await waitForI18n(comp);
+
+    const purchaseCard = comp.getByRole("article", { name: "Contact sales" });
+    expect(purchaseCard.dataset.ouiaComponentId).toBe(
+      "card-overview-contact-sales"
+    );
+
+    const link = comp.getByRole("link", { name: "Contact sales" });
+    expect(link.dataset.ouiaComponentId).toBe("link-contact-sales");
+  });
```

Comment:
```suggestion
  it("should persist ouia ids", async () => {
    const comp = render(<Example />);
    await waitForI18n(comp);

    expect(comp.getByRole("article", { name: "Purchase now" }))
      .toHaveAttribute("data-ouia-component-id", "card-overview-purchase-now");
    expect(comp.getByRole("link", { name: "Purchase now" }))
      .toHaveAttribute("data-ouia-component-id", "link-purchase-now");
  
    expect(comp.getByRole("article", { name: "Contact sales" }))
      .toHaveAttribute("data-ouia-component-id", "card-overview-contact-sales");
    expect(comp.getByRole("link", { name: "Contact sales" }))
      .toHaveAttribute("data-ouia-component-id", "link-purchase-now");
  });
```

I haven't tested this, but I'd simplify the test like this. It's ok to test more than one thing in a single test. To test attributes there is a specific `toHaveAttribute` function you can use.